inbox/mailsync/backends/imap/generic.py

```diff
@@ -154,6 +155,8 @@ def __repr__(self) -> str:
 class FolderSyncEngine(InterruptibleThread):
     """Base class for a per-folder IMAP sync engine."""
 
+    global_lock = threading.BoundedSemaphore(1)
```

Comment:
It's not just a lock but a BoundedSemaphore because I might consider increasing it later to 2 or something else. Also sync-engine uses BoundedSemaphore everywhere else and I wanted to stay consistent.